### PR TITLE
Prevent the status bar from updating too quickly

### LINKED ISSUE #12084: Weird flashing at the bottom bar?
### Discussed in https://github.com/microsoft/vscode-cpptools/discussions/12079

In 1.19.4 and newer, the language status seems to be updating too quickly and causing a strobing effect.  We may need to set a minimum duration for how long we keep the status up.

### PR DESCRIPTION
**Pull request by bobbrow**
Fixes: #12084

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -3741,7 +3741,8 @@ export class DefaultClient implements Client {
             // without being opened because otherwise users may not realize that
             // the header had changed (unless they view source control differences).
             await vscode.window.showTextDocument(headerFormatUriAndRanges[0].uri, {
-                selection: headerReplaceEditRange, preserveFocus: false });
+                selection: headerReplaceEditRange, preserveFocus: false
```
Comment (bobbrow) [inline Extension/src/LanguageServer/client.ts:3747]: I didn't intend to format this file, but it looks like it needed it...

**Review by sean-mcmanus — DISMISSED** [review]
FYI, the isRunningCodeAnalysis also has the potential to contribute to the C/C++ busy indicator, but from what I can tell, that always takes long enough, so it doesn't seem to need an explicit delay unless you think we should safe guard against that case anyway.

File: Extension/src/LanguageServer/dataBinding.ts [hunk 2]
```diff
@@ -4,13 +4,37 @@
...
     private isActive: boolean = true;
+    private deferral?: Deferral;
 
-    constructor(value: T) {
-        this.value = value;
+    /**
+     * Bind an event to a value so that a data model can automatically update the UI when values change.
+     * Since values can change quickly and cause UI to flicker, an optional delay/trigger combination can
+     * be specified to prevent UI elements from appearing/disappearing too quickly.
+     * @param value The initial value in the binding
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/dataBinding.ts]: This is missing a period while the other doc comments have them.

**Review by Colengms — CHANGES_REQUESTED** [review]
Can we remove the existing 1s timers (in the callbacks for some of these events)?

**Review by Colengms — DISMISSED** [review]

**Review by sean-mcmanus — APPROVED** [review]

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
Your latest change appears to "regress" the scenario where the icons are pinned compared to 1.19.6. I think the regression is caused by the removal of the existing timers (your original change didn't repro the issue), i.e. the result is that the pinned status text changes pretty fast, compared to slowly in 1.19.6.

Should the text change be changed back to the 1s timer as 1.19.6 -- the visual result looks better to me.

**Review by sean-mcmanus — CHANGES_REQUESTED** [review]
@bobbrow I'm seeing IntelliSense status being stuck at "Ready".

**Review by sean-mcmanus — APPROVED** [review]

**Review by Colengms — APPROVED** [review]

**Pull request merged**